p2p/peer.go

```diff
@@ -82,12 +83,13 @@ const (
 )
 
 // PeerEvent is an event emitted when peers are either added or dropped from
-// a p2p.Server
+// a p2p.Server or when a message is sent or received on a peer connection
 type PeerEvent struct {
```

Comment:
I think we should focus on writing the simulations that require these message events and then modify when we have a better idea of the requirements.